File: BilDunya/src/main/java/com/bildunya/dto/CreateContentRequest.java
package com.bildunya.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateContentRequest {

    @NotBlank(message = "Description is required")
    private String description;

    @NotNull(message = "Content type is required")
    private String contentType;

    @NotNull(message = "Latitude is required")
    private Double latitude;

    @NotNull(message = "Longitude is required")
    private Double longitude;

    private String locationName;
    private String shareType; // PUBLIC, PRIVATE, ANONYMOUS
    private String tags;
}

File: bildunya_frontend/lib/data/models/create_content_request.dart
/// [com.bildunya.dto.CreateContentRequest] — Jackson camelCase.
class CreateContentRequest {
  const CreateContentRequest({
    required this.description,
    required this.contentType,
    required this.latitude,
    required this.longitude,
    this.locationName,
    this.shareType,
    this.tags,
  });

  final String description;
  final String contentType;
  final double latitude;
  final double longitude;
  final String? locationName;
  final String? shareType;
  final String? tags;

  Map<String, dynamic> toJson() => {
        'description': description,
        'contentType': contentType,
        'latitude': latitude,
        'longitude': longitude,
        if (locationName != null) 'locationName': locationName,
        if (shareType != null) 'shareType': shareType,
        if (tags != null) 'tags': tags,
      };
}
